File: apple/Sources/ReactNativeWatchOS/Modifiers/Redacted.swift
import SwiftUI

/// SwiftUI `.redacted(reason:)`. Renders the subtree as a redacted
/// placeholder (skeleton / loading state). `RedactionReasons.placeholder`
/// is the only public reason on watchOS, so any value maps to it.
enum RNWRedactedModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("redacted") { view, _, _ in
            AnyView(view.redacted(reason: .placeholder))
        }
    }
}

File: src/modifiers/redacted.ts
import { createModifier } from './createModifier';

/// SwiftUI `RedactionReasons`. Only `'placeholder'` is expressible on
/// watchOS today (it's the single public reason in the framework).
export type RedactionReason = 'placeholder';

/// SwiftUI `.redacted(reason:)`. Renders the subtree as a redacted
/// placeholder (e.g. for skeleton / loading states). Defaults to
/// `'placeholder'`.
export function redacted(reason: RedactionReason = 'placeholder') {
  return createModifier('redacted', { reason });
}
